Beef up docs around using class fields (#516)
lit/lit.dev: packages/lit-dev-content/site/docs/components/decorators.md

To use decorators with [TypeScript](https://www.typescriptlang.org/docs/handbook/decorators.html), enable the `experimentalDecorators` compiler option.

> You should also ensure that the `useDefineForClassFields` setting is `false`. Note, this should only be required when the `target` is set to `es2020` or greater, but it's recommended to explicitly ensure this setting is `false`.

Reviewer: I believe it's only for `esnext`, not `es2020` (https://github.com/microsoft/TypeScript/issues/34787)

Author: Fixed

```json
"experimentalDecorators": true,
"useDefineForClassFields": false,
```

Enabling `emitDecoratorMetadata` is not required and not recommended.

### Using decorators with Babel  { #decorators-babel }

> If you're compiling JavaScript with [Babel](https://babeljs.io/docs/en/), you can enable decorators by adding the following plugins and assumptions:

Reviewer 2: Using "assumptions" here is super jargony. You'd have to be pretty well-versed in Babel to know that there are config options called  "Compiler assumptions," and the average user  is likely to infer the generic meaning for assumptions here.

Suggest either just go with "plugins" and assume (heh) that the configuration steps that follow are part of adding the plugins, or say, "plugins and settings"

Author: Changed to "plugins and settings"

*   [`@babel/plugin-proposal-decorators`](https://babeljs.io/docs/en/babel-plugin-proposal-decorators)
*   [`@babel/plugin-proposal-class-properties`](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties)

Note, the `@babel/plugin-proposal-class-properties` may not be required with the latest versions of Babel.

> To enable these settings, add code like this to your Babel configuration:

Reviewer 2: "these settings" doesn't really refer back to anything.

```suggestion
To set up the plugins, add code like this to your Babel configuration:
```
